File: modules/wem-api/src/main/java/com/enonic/wem/api/content/page/LayoutTemplate.java
package com.enonic.wem.api.content.page;


public class LayoutTemplate
    extends Template<LayoutTemplateId, LayoutTemplateName>
{
    private LayoutTemplate( final Builder builder )
    {
        super( builder.name, builder.id, builder.displayName, builder.descriptor, builder.config );
    }

    public static LayoutTemplate.Builder newLayoutTemplate()
    {
        return new Builder();
    }

    public static class Builder
        extends BaseTemplateBuilder<Builder, LayoutTemplateId, LayoutTemplateName>
    {
        private Builder()
        {
        }

        public LayoutTemplate build()
        {
            return new LayoutTemplate( this );
        }
    }
}

File: modules/wem-webapp/src/main/webapp/admin/common/js/content/page/LayoutTemplate.ts
module api_content_page{

    export class LayoutTemplate extends Template<LayoutTemplateName> {

        constructor(builder:LayoutTemplateBuilder) {
            super(builder);
        }
    }

    export class LayoutTemplateBuilder extends TemplateBuilder<LayoutTemplateName> {

        public build():Template<LayoutTemplateName> {
            return new LayoutTemplate(this);
        }
    }
}